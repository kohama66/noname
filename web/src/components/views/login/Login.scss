@import "src/css/common.scss";
#login {
  background-color: $baseColor;
  min-height: 100vh;
  box-sizing: border-box;
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  form {
    width: 300px;
    border-radius: 15px;
    @include whiteShadow;
    background-color: #fff;
    margin: 50px auto 0;
    box-sizing: border-box;
    padding: 50px 30px;
    label {
      color: $blackFontColor;
      font-size: 14px;
      font-weight: bold;
    }
    input {
      margin: 10px 0 20px;
      &:last-of-type {
        margin: 0px 0 0;
      }
    }
    a {
      font-size: 10px;
      color: $lightGray;
      text-decoration: none;
      text-align: center;
      display: block;
      width: 100%;
      margin: 10px 0 0;
    }
    .error-message {
      margin-top: 20px;
      height: 15px;
      padding: 5px 0;
      font-size: 12px;
      line-height: 18px;
      font-weight: bold;
      color: red;
    }
  }
}
@media screen and (max-width: 767px) {
  #login {
    padding: 120px 5%;
    form {
      width: 100%;
      margin: 30px auto 0;
      padding: 40px 20px;
      label{
        font-size: 12px;
      }
    }
  }
}
