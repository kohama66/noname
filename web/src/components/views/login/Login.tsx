import React, { FC, useContext, useState } from 'react';
import { Link, useHistory } from 'react-router-dom';
import { getGuest } from '../../../package/api';
import { signin } from '../../../package/api/auth';
import { deleteAuthToken } from '../../../utils/function/Cookie';
import { GuestContext } from '../../../utils/context/GuestContext';
import Title from '../guest/parts/Title/Title';
import Input from '../parts/formParts/Input';
import './Login.scss'
import { useError } from '../../../utils/hooks/Error';

const Login: FC = () => {
  const [email, setEmail] = useState<string>("")
  const [password, setPassword] = useState<string>("")
  const [disabled, setDisabled] = useState<boolean>(false)
  const { setGuest } = useContext(GuestContext)
  const { error, customError } = useError()
  const history = useHistory()

  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>): Promise<void> => {
    e.preventDefault()
    setDisabled(true)
    try {
      await signin({ email, password })
      setGuest((await getGuest()).guest)
      history.push("/guest")
    } catch (error) {
      deleteAuthToken()
      console.log(error.message)
      customError(error.message)
    }
    setDisabled(false)
  }

  return (
    <div id="login">
      <Title title="LOGIN" text="ログイン" />
      <form onSubmit={handleSubmit}>
        <label>メールアドレス</label>
        <Input type="email" value={email} setState={setEmail} required={true} maxLength={50} disabled={disabled} />
        <label>パスワード</label>
        <Input type="password" value={password} setState={setPassword} required={true} minLength={8}
          placeHolder="8文字以上で入力してください" disabled={disabled} />
        <div className="error-message">
          <p>{error}</p>
        </div>
        <Input type="submit" value="ログイン" disabled={disabled} />
        <Link to="guest/signup" >未登録の方はこちら</Link>
      </form>
    </div>
  )
}

export default Login;