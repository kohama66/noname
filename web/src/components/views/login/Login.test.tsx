import * as React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';
import {
  BrowserRouter as Router,
} from "react-router-dom";


it("login snapshot", () => {
  const tree = renderer.create(
    <Router>
      <Login />
    </Router>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})