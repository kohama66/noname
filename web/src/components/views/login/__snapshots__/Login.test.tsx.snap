// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1 1`] = `null`;

exports[`login snapshot 1`] = `
<div
  id="login"
>
  <div
    className="heading"
  >
    <h2>
      LOGIN
    </h2>
    <p>
      ログイン
    </p>
  </div>
  <form
    onSubmit={[Function]}
  >
    <label>
      メールアドレス
    </label>
    <input
      disabled={false}
      maxLength={50}
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <label>
      パスワード
    </label>
    <input
      disabled={false}
      minLength={8}
      onChange={[Function]}
      placeholder="8文字以上で入力してください"
      required={true}
      type="password"
      value=""
    />
    <div
      className="error-message"
    >
      <p />
    </div>
    <input
      disabled={false}
      onChange={[Function]}
      type="submit"
      value="ログイン"
    />
    <a
      href="/guest/signup"
      onClick={[Function]}
    >
      未登録の方はこちら
    </a>
  </form>
</div>
`;
